consensus/scheduled_task_test.go

```diff
@@ -0,0 +1,48 @@
+package consensus
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestScheduledTask(t *testing.T) {
+
+	// Test short scheduled task
+	{
+		task := NewScheduledTask[uint64]()
+
+		executedTaskParam := uint64(0)
+		task.Schedule(time.Microsecond, 100, func(param uint64) {
+			executedTaskParam = param
+		})
+		time.Sleep(time.Second)
```

Comment:
Should we reduce the sleep to something shorter?

Reply:
I'm down w/ 0.5 seconds.

Reply:
Sounds good